packages/api/src/tasks/checkAlerts.ts

```diff
@@ -497,7 +497,11 @@ export const processAlert = async (now: Date, alert: AlertDocument) => {
             windowSizeInMins,
           });
           history.counts += 1;
+          // will overwrite if multiple alerts fire
+          history.lastValue = totalCount;
         }
+        // only write one lastValue if there are no alerts at all?
+        history.lastValue ??= totalCount;
```

Comment:
This code is equal to one line `history.lastValue = totalCount;` outside of if condition. Now when I think about it, I think we probably want to put `{ bucketStart, totalCount}` in an array since history only represent the most recent checking point instead of bucket by bucket if that makes sense

Reply:
Sure, I like that better, what about `lastValues: {startTime: DateString, count: number}` as a signature? then we can do`lastValues.push({startTime: bucketStart, count: totalCount})`

Reply:
sounds good. I assume you meant `lastValues: {startTime: DateString, count: number}[]`